core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java

```diff
@@ -0,0 +1,96 @@
+package org.opentosca.toscana.plugins.kubernetes;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.opentosca.toscana.core.BaseSpringTest;
+import org.opentosca.toscana.core.csar.Csar;
+import org.opentosca.toscana.core.parse.CsarParseService;
+import org.opentosca.toscana.core.parse.InvalidCsarException;
+import org.opentosca.toscana.core.testdata.TestCsars;
+import org.opentosca.toscana.core.testdata.TestPlugins;
+import org.opentosca.toscana.core.testdata.TestTransformationContext;
+import org.opentosca.toscana.core.transformation.TransformationContext;
+import org.opentosca.toscana.core.util.Preferences;
+import org.opentosca.toscana.plugins.model.InvalidDockerAppException;
+
+import org.assertj.core.util.Lists;
+import org.eclipse.winery.model.tosca.yaml.TServiceTemplate;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+import static org.opentosca.toscana.core.testdata.TestCsars.CSAR_YAML_VALID_DOCKER_SIMPLETASK;
+
+public class KubernetesPluginTest extends BaseSpringTest {
+    private static final Logger log = LoggerFactory.getLogger(KubernetesPluginTest.class);
+
+    private static TServiceTemplate tServiceTemplate;
+    private static KubernetesPlugin plugin;
+    @Autowired
+    private CsarParseService csarParser;
+    @Autowired
+    private Preferences preferences;
+    @Autowired
+    private TestTransformationContext testTransformationContext;
+    @Autowired
+    private TestCsars testCsars;
+
+    @Before
+    public void setUp() {
+        plugin = new KubernetesPlugin();
+    }
+
+    @Test
+    public void validTServiceTemplate() throws InvalidDockerAppException, FileNotFoundException, InvalidCsarException {
+        Csar csar = testCsars.getCsar(CSAR_YAML_VALID_DOCKER_SIMPLETASK);
+        tServiceTemplate = csarParser.parse(csar);
+        plugin.getDockerApp(tServiceTemplate);
+    }
+
+    @Test(expected = InvalidDockerAppException.class)
+    public void dockerAppMissing() throws FileNotFoundException, InvalidCsarException, InvalidDockerAppException {
+        Csar csar = testCsars.getCsar(TestCsars.CSAR_YAML_INVALID_DOCKERAPP_MISSING);
+        tServiceTemplate = csarParser.parse(csar);
+        plugin.getDockerApp(tServiceTemplate);
+    }
+
+    @Test
+    public void transformTest() throws Exception {
+        File artifactRootPath = new File(preferences.getDataDir(), "test-csar/transformations/testplatform_one/");
```

Comment:
Your test should not hard code the artifact path. When the repo implementation changes, your test breaks.
As your using the TestTransformationContext, I suggest you use:
```
@Autowired
TransformationDao transformationDao;
transformationDao.getContentDir()
```

Reply:
So you suggest changing the TestTransformationContext class?